great_expectations/core/usage_statistics/anonymizers/profiler_run_anonymizer.py

```diff
@@ -0,0 +1,205 @@
+
+        batch_request: Optional[dict] = parameter_builder.get("batch_request")
+        if batch_request:
+            anonymized_parameter_builder[
+                "anonymized_batch_request"
+            ] = self._batch_request_anonymizer.anonymize_batch_request(
+                *(), **batch_request
+            )
+            logger.debug("Anonymized batch request in ParameterBuilder")
+
+        return anonymized_parameter_builder
+
+    def _anonymize_expectation_configuration_builders(
+        self, expectation_configuration_builders: List[dict]
+    ) -> List[dict]:
+        anonymized_expectation_configuration_builders: List[dict] = []
+
+        for expectation_configuration_builder in expectation_configuration_builders:
+            anonymized_expectation_configuration_builder: dict = (
+                self._anonymize_expectation_configuration_builder(
+                    expectation_configuration_builder
+                )
+            )
+            anonymized_expectation_configuration_builders.append(
+                anonymized_expectation_configuration_builder
+            )
+
+        return anonymized_expectation_configuration_builders
+
+    def _anonymize_expectation_configuration_builder(
+        self, expectation_configuration_builder: dict
+    ) -> dict:
+        anonymized_expectation_configuration_builder: dict = {}
+
+        anonymized_expectation_configuration_builder[
+            "class_name"
+        ] = expectation_configuration_builder.get("class_name")
+        anonymized_expectation_configuration_builder[
+            "expectation_type"
+        ] = expectation_configuration_builder.get("expectation_type")
+
+        condition: Optional[str] = expectation_configuration_builder.get("condition")
+        if condition:
+            anonymized_expectation_configuration_builder[
+                "anonymized_condition"
+            ] = self.anonymize(condition)
+            logger.debug("Anonymized condition in ExpectationConfigurationBuilder")
+
+        return anonymized_expectation_configuration_builder
+
+    # noinspection PyUnusedLocal,PyUnresolvedReferences
+    @staticmethod
+    def resolve_config_using_acceptable_arguments(
+        profiler: "RuleBasedProfiler",  # noqa: F821
+        variables: Optional[Dict[str, Any]] = None,
+        rules: Optional[Dict[str, Dict[str, Any]]] = None,
+    ) -> dict:
+        runtime_config: dict = profiler.config.to_dict()
+
+        # If applicable, override config attributes with runtime args
+        if variables:
+            runtime_config["variables"] = variables
+        if rules:
+            runtime_config["rules"] = rules
+
+        # Ensure that the list of rules contains all relevant information (including rule names)
+        resolved_rules: List[dict] = []
+        for name, rule in runtime_config["rules"].items():
+            rule["name"] = name
+            resolved_rules.append(rule)
```

Comment:
In order to get the name of each rule, we need to ensure the resulting dictionary representation of the rule has it's name as a field. If we don't care about the anonymized name of the rule, we can forgo this.